### DISCUSSION TITLE
Can language_mapper be dropped off in a favor of vanilla Intl.DisplayNames

### DISCUSSION THREAD

**Discussion by zhuiks** [ezra-bible-app/ezra-bible-app#305] [Development]
Currently the most used function in  [`language_mapper`](https://github.com/ezra-project/ezra-project/blob/50a9051b0e45899f92daa3327c5b3bf5a1178a27/app/lib/language_mapper.js) is `getLanguageName(languageCode)`.
Language_mapper is based on `iso-639-3` packet.

The same functionality can be done with vanilla js [`Intl.DisplayNames`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames) as it's done in `i18nHelper` the refactored under `feature-locale-option` branch.

Is there any other use of `language_mapper`?

**Comment by tobias-klein** [comment]
If we can refactor this based on vanilla JS ... sure, why not!

The three public interface functions are:
- getLanguageCode (Gets the language code for a given language name)
- getLanguageName (Gets the language name for a given language code)
- mappingExists (Checks whether the code is mapped within the object list available via iso-639-3)

This is currently used in:
- One database migration (*updateTranslationLanguageCodes.js)
- BibleTranslation database model (deprecated, but still existing for compatibility when upgrading from older Ezra versions)
- InstallModuleAssistant
- RemoveModuleAssistant
- SwordModuleHelper

This is currently not a priority to me, but if you would like to take it up - feel free to do so!

**Comment by tobias-klein** [comment]
Btw. I moved this to the Development category, because I would like to reserve Q&A for user questions.

**Comment by zhuiks** [comment]
I'm humbled.
Bible translators got much farther than JavaScript!

Vanilla JavaScript works so far only with ISO 639-1 and 639-2 language codes. The most complete list of languages is under ISO 639-3. 
`languageMapper` is basically a wrapper around the external library that uses ISO 639-3
